### ISSUE TITLE
`RegexpError: invalid multibyte escape` when using JRuby 9.3.3.0

### ISSUE THREAD

**Issue by koic** [jruby/jruby#7113]
**Environment Information**
  
Provide at least:
* JRuby version (`jruby -v`) and command line (flags, JRUBY_OPTS, etc)

`jruby 9.3.3.0 (2.6.8) 2022-01-19 b26de1f5c5 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]`

* Operating system and platform (e.g. `uname -a`)

`Darwin atelier.local 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64`

**Expected Behavior**

There is no error like CRuby. e.g:

```console
% ruby -v
ruby 2.5.9p229 (2021-04-05 revision 67939) [x86_64-darwin19]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"

% ruby -v
ruby 2.6.9p207 (2021-11-24 revision 67954) [x86_64-darwin19]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"

% ruby -v
ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-darwin19]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"

% ruby -v
ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-darwin19]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"

% ruby -v
ruby 3.1.1p18 (2022-02-18 revision 53f5fc4236) [x86_64-darwin19]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"
```

**Actual Behavior**

`RegexpError: invalid multibyte escape: /\x82/` error will occur.

```console
% ruby -v
jruby 9.3.3.0 (2.6.8) 2022-01-19 b26de1f5c5 Java HotSpot(TM) 64-Bit Server VM 25.271-b09 on 1.8.0_271-b09 +jit [darwin-x86_64]
% ruby -e "Regexp.new('\x82', Regexp::NOENCODING)"
RegexpError: invalid multibyte escape: /\x82/
  initialize at org/jruby/RubyRegexp.java:956
         new at org/jruby/RubyClass.java:893
      <main> at -e:1
```

I encountered this issue in RuboCop's CI build.
https://app.circleci.com/pipelines/github/rubocop/rubocop/5866/workflows/c9caa01a-1bb0-4bbb-893e-74e6404646f4/jobs/210977

**Comment by gengle** [comment]
I'd like to leave this pull request open until decisions have been captured relating to these classes.  I'd prefer handling everything in this one PR instead of chaining  things together for further review.

**Comment by headius** [comment]
@gengle What PR are you referring to?

**Comment by koic** [comment]
This issue was resolved in JRuby 9.3.9.0. Thank you!
